internal/backend/rest/rest.go

```diff
@@ -246,6 +247,20 @@ func (b *Backend) openReader(ctx context.Context, h restic.Handle, length int, o
 		return nil, errors.Errorf("unexpected HTTP response (%v): %v", resp.StatusCode, resp.Status)
 	}
 
+	// workaround https://github.com/golang/go/issues/46071
+	// see also https://forum.restic.net/t/http2-stream-closed-connection-reset-context-canceled/3743/10
+	if resp.ContentLength == 0 && resp.ProtoMajor == 2 && resp.ProtoMinor == 0 {
+		if clens := resp.Header["Content-Length"]; len(clens) == 1 {
+			if cl, err := strconv.ParseUint(clens[0], 10, 63); err == nil {
```

Comment:
How about returning an error to the caller if the value of the `Content-Length` header cannot be parsed? If it is present but the value is invalid, that's an error. Otherwise, users will just see the "unexpected EOF" error when the server returns garbage in the header.

Reply:
If the golang client is unable to parse `Content-Length`, then `resp.ContentLength` is probably equal to `-1`. And in that case our error handling will be dead code.

Based on the same argument, it should probably be safe to use `resp.Header.Get("Content-Length")`

Reply:
And to immediately contradict myself: Obviously the content-length was not parsed yet, because otherwise it wouldn't be zero...